util/network-devp2p/src/connection.rs

```diff
@@ -360,51 +357,50 @@ impl EncryptedConnection {
 		if len > MAX_PAYLOAD_SIZE {
 			bail!(ErrorKind::OversizedPacket);
 		}
+
 		header.append_raw(&[(len >> 16) as u8, (len >> 8) as u8, len as u8], 1);
 		header.append_raw(&[0xc2u8, 0x80u8, 0x80u8], 1);
-		//TODO: get rid of vectors here
+		let padding = (16 - (len % 16)) % 16;
+
+		let mut packet = vec![0u8; 32 + len + padding + 16];
 		let mut header = header.out();
-		let padding = (16 - (payload.len() % 16)) % 16;
 		header.resize(16, 0u8);
-
-		let mut packet = vec![0u8; 32 + payload.len() + padding + 16];
-		self.encoder.encrypt(&mut RefReadBuffer::new(&header), &mut RefWriteBuffer::new(&mut packet), false).expect("Invalid length or padding");
-		EncryptedConnection::update_mac(&mut self.egress_mac, &mut self.mac_encoder,  &packet[0..16]);
-		self.egress_mac.clone().finalize(&mut packet[16..32]);
-		self.encoder.encrypt(&mut RefReadBuffer::new(payload), &mut RefWriteBuffer::new(&mut packet[32..(32 + len)]), padding == 0).expect("Invalid length or padding");
+		let header_len = header.len();
```

Comment:
it was resized the line above, so I would think it is 16 always (we could use a constant for it).